pkg/vsphere/simulator/registry.go

```diff
@@ -105,3 +111,31 @@ func (r *Registry) Remove(item types.ManagedObjectReference) {
 
 	delete(r.objects, item)
 }
+
+func (r *Registry) GetEntityParent(item mo.Entity, kind string) mo.Entity {
+	for {
+		parent := item.Entity().Parent
+
+		item = Map.Get(*parent).(mo.Entity)
+
+		if item.Reference().Type == kind {
+			return item
```

Comment:
Yes, it'll panic if parent == nil.  But that's only possible if there's a bug in the simulator.  User input can't produce such a state.